src/azure/cli/commands/arm.py

```diff
@@ -150,3 +151,201 @@ def required_values_validator(namespace):
     APPLICATION.remove(APPLICATION.COMMAND_TABLE_LOADED, add_id_parameters)
 
 APPLICATION.register(APPLICATION.COMMAND_TABLE_LOADED, add_id_parameters)
+
+def register_generic_update(name, getter, setter, setter_arg_name='parameters'):
+    from msrestazure.azure_operation import AzureOperationPoller
+    from azure.cli.commands import CliCommand, command_table
+    from azure.cli.commands._introspection import extract_args_from_signature
+
+    get_arguments = dict(extract_args_from_signature(getter))
+    set_arguments = dict(extract_args_from_signature(setter))
+
+    def handler(args):
+        ordered_arguments = args.pop('ordered_arguments')
+
+        getterargs = {key: val for key, val in args.items()
+                      if key in get_arguments}
+        instance = getter(**getterargs)
+
+        # Update properties
+        for arg in ordered_arguments:
+            arg_type, expressions = arg
+            if arg_type == '--set':
+                try:
+                    for expression in expressions:
+                        set_properties(instance, expression)
+                except ValueError:
+                    raise CLIError('--set should be of the form:'
+                                   ' --set property.property=<value>'
+                                   ' property2.property=<value>')
+            elif arg_type == '--add':
+                try:
+                    add_properties(instance, expressions)
+                except ValueError:
+                    raise CLIError('--add should be of the form:'
+                                   ' --add property.list key1=value1 key2=value2')
+            elif arg_type == '--remove':
+                try:
+                    remove_properties(instance, expressions)
+                except ValueError:
+                    raise CLIError('--remove should be of the form: --remove'
+                                   ' property.propertyToRemove or'
+                                   ' --remove property.list <indexToRemove>')
+            else:
+                raise ValueError('Unsupported arg type {}'.format(arg_type))
```

Comment:
this would only happen if we added or renamed one of the parameters.  I added this check since we're depending directly on the string names of the params to index on.  Without this check, the code would silently continue and do nothing if the names ever changed

---
In reply to: [72831907](https://github.com/Azure/azure-cli/pull/558#discussion_r72831907) [](ancestors = 72831907)